test/system/010-images.bats

```diff
@@ -359,4 +359,36 @@ EOF
     run_podman --root $imstore/root rmi --all
 }
 
+@test "podman images with concurrent removal" {
+    skip_if_remote "following test is not supported for remote clients"
+    local count=5
+
+    # First build $count images
+    for i in $(seq --format '%02g' 1 $count); do
+        cat >$PODMAN_TMPDIR/Containerfile <<EOF
+FROM $IMAGE
+RUN echo $i
+EOF
+        run_podman build -q -t i$i $PODMAN_TMPDIR
+    done
+
+    run_podman images
+    # Now remove all images in parallel and in the background and make sure
+    # that listing all images does not fail (see BZ 2216700).
+    for i in $(seq --format '%02g' 1 $count); do
+        echo "REMOVING IMAGE i$i"
+        timeout --foreground -v --kill=10 60 \
+                $PODMAN rmi i$i &
+    done
+
+    # Now keep on listing images until all created ones have been removed and
+    # only one (i.e., $IMAGE) is left.
+    while [[ ${#lines[*]} -gt 1 ]]; do
+        run_podman images -q
+    done
```

Comment:
What if one rmi fails, then this would loop forever , no?

Reply:
Yes, that would loop forever.  I don't have an idea how to prevent that.  It would be easier to handle in the e2e tests but I really want to have a regression test in gating.